src/home_robot/home_robot/agent/imagenav_agent/visualizer.py

```diff
@@ -0,0 +1,640 @@
+
+        sem_img = cv2.cvtColor(sem_img, cv2.COLOR_RGB2BGR)
+        sem_img = self._add_border(sem_img, border_size)
+        w = sem_img.shape[1]
+
+        top_bar = np.ones((text_bar_height, w, 3), dtype=np.uint8) * 255
+        frame = np.concatenate([top_bar, sem_img.astype(np.uint8)], axis=0)
+
+        font = cv2.FONT_HERSHEY_SIMPLEX
+        fontScale = 0.8
+        color = (20, 20, 20)
+        thickness = 2
+
+        text = "Observation"
+        textsize = cv2.getTextSize(text, font, fontScale, thickness)[0]
+        textX = (w - textsize[0]) // 2
+        textY = (text_bar_height + border_size + textsize[1]) // 2
+        frame = cv2.putText(
+            frame,
+            text,
+            (textX, textY),
+            font,
+            fontScale,
+            color,
+            thickness,
+            cv2.LINE_AA,
+        )
+        return frame
+
+    def make_map_preds(
+        self,
+        sensor_pose,
+        obstacle_map,
+        explored_map,
+        semantic_map,
+        closest_goal_map,
+        goal_map,
+        visualize_goal,
+    ):
+        
+        if semantic_map is None:
+            fill_val = self.num_sem_categories - 1
+            semantic_map = np.zeros_like(obstacle_map) + fill_val
+
+        curr_x, curr_y, curr_o, gy1, gy2, gx1, gx2 = sensor_pose
+        gy1, gy2, gx1, gx2 = int(gy1), int(gy2), int(gx1), int(gx2)
+
+        # Update visited map with last visited area
+        if self.last_xy is not None:
+            last_x, last_y = self.last_xy
+            last_pose = [
+                int(last_y * 100.0 / self.map_resolution - gy1),
+                int(last_x * 100.0 / self.map_resolution - gx1),
+            ]
+            last_pose = pu.threshold_poses(last_pose, obstacle_map.shape)
+            curr_pose = [
+                int(curr_y * 100.0 / self.map_resolution - gy1),
+                int(curr_x * 100.0 / self.map_resolution - gx1),
+            ]
+            curr_pose = pu.threshold_poses(curr_pose, obstacle_map.shape)
+            self.visited_map_vis[gy1:gy2, gx1:gx2] = draw_line(
+                last_pose, curr_pose, self.visited_map_vis[gy1:gy2, gx1:gx2]
+            )
+        self.last_xy = (curr_x, curr_y)
+
+        semantic_map += 6
+
+        # Obstacles, explored, and visited areas
+        no_category_mask = semantic_map == 6 + self.num_sem_categories - 1
+        obstacle_mask = np.rint(obstacle_map) == 1
+        explored_mask = np.rint(explored_map) == 1
+        visited_mask = self.visited_map_vis[gy1:gy2, gx1:gx2] == 1
+        semantic_map[no_category_mask] = 0
+        semantic_map[np.logical_and(no_category_mask, explored_mask)] = 2
+        semantic_map[np.logical_and(no_category_mask, obstacle_mask)] = 1
+        semantic_map[visited_mask] = 3
+
+        # Goal
+        if visualize_goal:
+            selem = skimage.morphology.disk(4)
+            goal_mat = 1 - skimage.morphology.binary_dilation(goal_map, selem) != True
```

Comment:
@theophilegervet i have no idea what this is doing but the linter does not like it